nifi-api/src/main/java/org/apache/nifi/authorization/AccessPolicy.java

```diff
@@ -0,0 +1,291 @@
+        return Objects.hashCode(this.identifier);
+    }
+
+    @Override
+    public String toString() {
+        return String.format("identifier[%s], resource[%s], entityId[%s], action[%s]",
+                getIdentifier(), getResource().getIdentifier(), getEntities(), getActions(), ", ");
+    }
+
+    /**
+     * Builder for Access Policies.
+     */
+    public static class AccessPolicyBuilder {
+
+        private String identifier;
+        private Resource resource;
+        private Set<String> entities = new HashSet<>();
+        private Set<RequestAction> actions = new HashSet<>();
+        private final boolean fromPolicy;
+
+        /**
+         * Default constructor for building a new AccessPolicy.
+         */
+        public AccessPolicyBuilder() {
+            this.fromPolicy = false;
+        }
+
+        /**
+         * Initializes the builder with the state of the provided policy. When using this constructor
+         * the identifier field of the builder can not be changed and will result in an IllegalStateException
+         * if attempting to do so.
+         *
+         * @param other the existing access policy to initialize from
+         */
+        public AccessPolicyBuilder(final AccessPolicy other) {
+            if (other == null) {
+                throw new IllegalArgumentException("Can not initialize builder with a null access policy");
+            }
+
+            this.identifier = other.getIdentifier();
+            this.resource = other.getResource();
+            this.entities.clear();
+            this.entities.addAll(other.getEntities());
+            this.actions.clear();
+            this.actions.addAll(other.getActions());
+            this.fromPolicy = true;
+        }
+
+        /**
+         * Sets the identifier of the builder.
+         *
+         * @param identifier the identifier to set
+         * @return the builder
+         * @throws IllegalStateException if this method is called when this builder was constructed from an existing Policy
+         */
+        public AccessPolicyBuilder identifier(final String identifier) {
+            if (fromPolicy) {
+                throw new IllegalStateException(
+                        "Identifier can not be changed when initialized from an existing policy");
+            }
+
+            this.identifier = identifier;
+            return this;
+        }
+
+        /**
+         * Sets the resource of the builder.
+         *
+         * @param resource the resource to set
+         * @return the builder
+         */
+        public AccessPolicyBuilder resource(final Resource resource) {
+            this.resource = resource;
+            return this;
+        }
+
+        /**
+         * Adds all the entities from the provided set to the builder's set of entities.
+         *
+         * @param entities the entities to add
+         * @return the builder
+         */
+        public AccessPolicyBuilder addEntities(final Set<String> entities) {
```

Comment:
We probably could do that... personally I like being able to say addEntity("a").addEntity("b")  rather than creating a set of 2 elements, but I'm sure theres some utility that returns a set from var args and could make it just as easy.

Since the methods are already there I would opt to leave them unless you felt really strongly about it.